biz.aQute.bnd.remote.junit/src/aQute/bnd/remote/junit/BundleBuilder.java

```diff
@@ -36,47 +40,43 @@ public BundleBuilder x() {
 
 	Entry<String, Map<String, String>> add(String s, Entry<String, Map<String, String>> entry, String key,
 			String value) {
-		if (entry == null)
-			throw new IllegalArgumentException("Requires " + s + " to be set before adding " + key + "=" + value);
+		checkNonNullArgument(s, "'s' cannot be null");
+		checkNonNullArgument(entry, "Requires " + s + " to be set before adding " + key + "=" + value);
+		checkNonNullArgument(key, "Key cannot be null");
+		checkNonNullArgument(value, "Value cannot be null");
 
-		String put = entry.getValue()
-				.put(key, value);
-		if (put != null)
-			throw new IllegalArgumentException(
-					"Value already set " + s + " : " + key + "=" + value + " old value was: " + put);
+		String put = entry.getValue().put(key, value);
+		checkState(put != null, "Value already set " + s + " : " + key + "=" + value + " old value was: " + put);
 		return entry;
 	}
 
 	void add(String header, Map<String, Map<String, String>> s, String outerkey,
 			String innerKey, String value) {
-
-		Map<String, String> map = s.get(outerkey);
-		if (map == null) {
-			map = new LinkedHashMap<>();
-			s.put(outerkey, map);
-		}
+		Map<String, String> map = s.computeIfAbsent(outerkey, k -> new LinkedHashMap<>());
+		s.put(outerkey, map);
 		map.put(innerKey, value);
 	}
 
 	String join(String... packageNames) {
-		return Stream.of(packageNames)
-				.collect(Collectors.joining(","));
+		checkNonNullArgument(packageNames, "'packageNames' cannot be null");
+		return Stream.of(packageNames).collect(Collectors.joining(","));
 	}
 
 	String prepare(String name, Map<String, Map<String, String>> domain) {
+		StringBuilder builder = new StringBuilder(name);
 		while (domain.containsKey(name)) {
-			name += "~";
+			builder.append("~");
 		}
-		domain.put(name, new LinkedHashMap<>());
-		return name;
+		domain.put(builder.toString(), new LinkedHashMap<>());
```

Comment:
Just call builder.toString() once. use a var to hold it for the 2 uses of the string. (Avoid rework).